sympy/concrete/tests/test_sums_products.py

```diff
@@ -572,23 +572,24 @@ def test_Sum_doit():
     s = Sum( Sum( Sum(2,(z,1,n+1)), (y,x+1,n)), (x,1,n))
     assert 0 == (s.doit() - n*(n+1)*(n-1)).factor()
 
-    assert Sum(KroneckerDelta(m, n), (m, -oo, oo)).doit() == Piecewise((1, And(-oo < n, n < oo)), (0, True))
-    assert Sum(x*KroneckerDelta(m, n), (m, -oo, oo)).doit() == Piecewise((x, And(-oo < n, n < oo)), (0, True))
+    # Integer assumes finite
+    # assert Sum(KroneckerDelta(m, n), (m, -oo, oo)).doit() == Piecewise((1, And(-oo < n, n < oo)), (0, True))
```

Comment:
```suggestion
    assert Sum(KroneckerDelta(x, y), (x, -oo, oo)).doit() == Piecewise((1, And(-oo < y, y < oo)), (0, True))
```

Reply:
```
In [8]: print(Sum(KroneckerDelta(x, y), (x, -oo, oo)).doit())
Piecewise((1, (y >= -oo) & (y < oo)), (0, True))
```

so, no. I'm actually not that keen on adding the "correct" expression either. (Nor fixing the issue so that it returns what may be expected.)

```
In [13]: y = Symbol('y', extended_real=True)

In [14]: print(Sum(KroneckerDelta(x, y), (x, -oo, oo)).doit())
Piecewise((1, y < oo), (0, True))
```